lib/taurus/qt/qtcore/util/emitter.py

```diff
@@ -184,7 +216,22 @@ def __init__(self, parent=None, name='', queue=None, method=None, cursor=None, s
         self._done = 0
         # Moved to the end to prevent segfaults ...
         self.emitter.doSomething.connect(self._doSomething)
-        self.emitter.somethingDone.connect(self.next)
+        
+        if not self.refreshTimer:
+            self.emitter.somethingDone.connect(self.next)
+        
+        
+    def onRefresh(self):
+        try:
+            self.refreshTimer.setInterval(self.polling)
```

Comment:
This seems **very** ugly to me. You are resetting the interval of the timer **on each callback from the timer**